subsys/sensing/sensor/hinge_angle/hinge_angle.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sensing/sensing_sensor.h>
+
+LOG_MODULE_REGISTER(hinge_angle, CONFIG_SENSING_LOG_LEVEL);
+
+#define HINGE_REPORTER_NUM 2
+
+static struct sensing_sensor_register_info hinge_reg = {
+	.flags = SENSING_SENSOR_FLAG_REPORT_ON_CHANGE,
+	.sample_size = sizeof(struct sensing_sensor_value_q31),
+	.sensitivity_count = 1,
+	.version.value = SENSING_SENSOR_VERSION(1, 0, 0, 0),
+};
+
+struct hinge_angle_context {
+	struct rtio_iodev_sqe *sqe;
+	sensing_sensor_handle_t reporters[HINGE_REPORTER_NUM];
+	struct sensing_sensor_value_3d_q31 sample[HINGE_REPORTER_NUM];
+	int has_sample[HINGE_REPORTER_NUM];
+};
+
+static int hinge_init(const struct device *dev)
+{
+	struct hinge_angle_context *data = dev->data;
+	int ret;
+
+	ret = sensing_sensor_get_reporters(dev,
+			SENSING_SENSOR_TYPE_MOTION_ACCELEROMETER_3D,
+			data->reporters, HINGE_REPORTER_NUM);
+	if (ret != HINGE_REPORTER_NUM) {
+		LOG_ERR("%s: reporter mismatch:%d", dev->name, ret);
+		return -ENODEV;
+	}
+
+	LOG_INF("%s:Found reporter 0: %s", dev->name,
+			sensing_get_sensor_info(data->reporters[0])->name);
+	LOG_INF("%s:Found reporter 1: %s", dev->name,
+			sensing_get_sensor_info(data->reporters[1])->name);
+
+	return 0;
+}
+
+static int hinge_attr_set(const struct device *dev,
+		enum sensor_channel chan,
+		enum sensor_attribute attr,
+		const struct sensor_value *val)
+{
+	struct sensing_sensor_config config = {0};
+	struct hinge_angle_context *data = dev->data;
+	int ret = 0;
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		config.attri = SENSING_SENSOR_ATTRIBUTE_INTERVAL;
+		config.interval = (uint32_t)(USEC_PER_SEC * 1000LL /
+				sensor_value_to_milli(val));
+		ret = sensing_set_config(data->reporters[0], &config, 1);
+		ret |= sensing_set_config(data->reporters[1], &config, 1);
+		break;
+
+	case SENSOR_ATTR_HYSTERESIS:
+		break;
+
+	default:
+		ret = -ENOTSUP;
+		break;
+	}
+
+	LOG_INF("%s set attr:%d ret:%d", dev->name, attr, ret);
+	return ret;
+}
+
+static int hinge_submit(const struct device *dev,
```

Comment:
This is an infallible call and should be void hinge_submit(const struct device *dev, struct rtio_iodev_sqe *sqe);

Reply:
`hinge_submit` is an instance of `typedef int (*sensor_submit_t)(const struct device *sensor, struct rtio_iodev_sqe *sqe)`, so it should be `int`.

Reply:
Some cleanup needs to be done around this API it seems, thanks for pointing this out